Automattic/woocommerce-payments
# Add BannerNotice component and update FraudProtectionRuleCardNotice to use banner notices.

Fixes #5798 

#### Changes proposed in this Pull Request

This PR introduces a new component (`BannerNotice`) and updates the `FraudProtectionRuleCardNotice` component to make use of this new element. 

This component designs are outlined this Design Library: Kofpo8jj4zDWXDsVdlSz0f-fi-4172-49449

See this comment on the issue for screenshots of the designs: https://github.com/Automattic/woocommerce-payments/issues/5798#issuecomment-1473658749 

As part of this PR I've updated the notices introduced as part of https://github.com/Automattic/woocommerce-payments/pull/5638. @tpaksu I've included side-by-side screenshots for comparison below. I noticed that as part of your PR you included each icon SVG. As part of this PR I changed these to the use the `NoticeOutlineIcon`. This is slightly different to the SVG version. Let you know if you'd prefer I revert to use your SVGs - you may have discussed as a team the alternatives.

There is no equivalent for the info icon you need, so I've kept the info SVG you had. 

|<img width="849" alt="Screenshot 2023-03-20 at 4 38 02 pm" src="https://user-images.githubusercontent.com/8490476/226266100-a6aa2aca-3a18-460e-8131-b79d30c4da0e.png">|<img width="850" alt="Screenshot 2023-03-20 at 4 37 54 pm" src="https://user-images.githubusercontent.com/8490476/226266104-acbc8fd8-2e46-4283-bed0-4f5b0f9e4f79.png">|
| ----------- | ----------- |
|<img width="842" alt="Screenshot 2023-03-20 at 4 37 48 pm" src="https://user-images.githubusercontent.com/8490476/226266107-d44cce53-c279-4c83-a6ce-065c3741a79d.png">|<img width="851" alt="Screenshot 2023-03-20 at 4 37 39 pm" src="https://user-images.githubusercontent.com/8490476/226266111-347a9eeb-9bea-4669-a0da-d861500263db.png">|
|<img width="855" alt="Screenshot 2023-03-20 at 4 37 32 pm" src="https://user-images.githubusercontent.com/8490476/226266115-6fe2cdce-f175-483e-a27a-c1c6f551510b.png">|<img width="849" alt="Screenshot 2023-03-20 at 4 37 25 pm" src="https://user-images.githubusercontent.com/8490476/226266122-4863199a-a74a-4548-87e2-1cca9341e021.png">|
|<img width="845" alt="Screenshot 2023-03-20 at 4 37 13 pm" src="https://user-images.githubusercontent.com/8490476/226266125-a0c4e56c-159b-4442-b4b5-1f0b0f3eaf45.png">|<img width="854" alt="Screenshot 2023-03-20 at 4 36 59 pm" src="https://user-images.githubusercontent.com/8490476/226266129-e23aceda-f470-4e84-8a3e-f32474d4fe5e.png">|

#### Testing instructions

1. Checkout this branch.
2. Enable the fraud protection feature by running `npm run wp option update wcpay_fraud_protection_settings_active 1`
3. Go to WCPay Settings, and you should see the Fraud Protection section on the end of the form, before the Advanced link.
4. Check the "Advanced" option in the Fraud Protection section (see screenshot below). 
5. Check all the notice elements look correct. 
6. There are 2 hidden notices hidden on load (the error ones specifically). To show those:
   1. Go to the **"Purchase Price Threshold"** card
   1. Enable filtering
   1. Enter a minimum prices greater than the minimum.
   1. You should get the error notice show. 
   1. Go to the **"Order Items Threshold"** card
   1. Enable filtering
   1. Enter a minimum number of items greater than the maximum.
   1. You should get the error notice show. 

<img width="967" alt="Screenshot 2023-03-20 at 5 29 40 pm" src="https://user-images.githubusercontent.com/8490476/226273982-8340d7da-f21d-4216-b376-aca9953aa511.png">

-------------------


The notices on a mobile device viewport:
<img width="406" alt="Screenshot 2023-03-21 at 2 37 33 pm" src="https://user-images.githubusercontent.com/8490476/226518221-24cce34e-ce0a-41a0-987a-aabafadbf6aa.png">

**Post merge**
